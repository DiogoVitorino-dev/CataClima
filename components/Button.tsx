import {Pressable, StyleProp, StyleSheet, ViewStyle, useColorScheme} from 'react-native';
import {View} from '../components/Themed';
import {OpenText} from './StyledText';
import Colors from '../constants/Colors';

export default function Button({
  label,
  disabled,
  onPress,
  style,
}: {
  label: string;  
  disabled?: boolean;
  onPress: Function;
  style?: StyleProp<ViewStyle>;
}) {
  const colorScheme = useColorScheme()

  return (
    
    <View 
      style={[
        styles.container,
        style,{
        borderColor:Colors[colorScheme ?? 'light'].borderColor,
        borderWidth:StyleSheet.hairlineWidth
      }
      ]}
      lightColor="#FAFAFA"
      darkColor="#1A1A1A">
      <Pressable 
        style={styles.button} 
        onPress={() => onPress()} disabled={disabled}>          
        {({pressed}) => (
          <OpenText 
            style={[styles.label, {opacity: pressed ? 0.5 : 1}]}>
            {label}
          </OpenText>
        )}
      </Pressable>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    maxHeight: 40,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 20,
    shadowColor: 'rgba(0,0,0,0.3)',
    shadowOffset: {width: 5, height: 5},
    shadowRadius: 10,
    elevation: 5,
  },

  button: {
    width: '100%',
    height: '100%',
    paddingHorizontal: 20,
    paddingVertical: 5,
    justifyContent: 'center',
    alignItems: 'center',
  },

  label: {
    textAlign: 'center',
    fontSize: 16,
  },
});
