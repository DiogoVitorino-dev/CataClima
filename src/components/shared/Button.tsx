import {Pressable, StyleProp, StyleSheet, ViewStyle, useColorScheme} from 'react-native';
import { View } from './Themed';
import { Colors } from '../../constants';
import { OpenTextStyled } from './StyledText';

export function Button({
	label,
	disabled,
	onPress,
	style,
}: {
  label: string;  
  disabled?: boolean;
  onPress: () => void;
  style?: StyleProp<ViewStyle>;
}) {
	const colorScheme = useColorScheme();

	return (
    
		<View 
			style={[
				styles.container,
				style,{
					borderColor:Colors[colorScheme ?? 'light'].borderColor,
					borderWidth:StyleSheet.hairlineWidth
				}
			]}
			lightColor="#FAFAFA"
			darkColor="#1A1A1A">
			<Pressable 
				style={styles.button}
				testID='pressableTestID'
				onPress={() => onPress()} disabled={disabled}>          
				{({pressed}) => (
					<OpenTextStyled 
						style={[styles.label, {opacity: pressed ? 0.5 : 1}]}>
						{label}
					</OpenTextStyled>
				)}
			</Pressable>
		</View>
	);
}

const styles = StyleSheet.create({
	container: {
		maxHeight: 40,
		justifyContent: 'center',
		alignItems: 'center',
		borderRadius: 20,
		shadowColor: 'rgba(0,0,0,0.3)',
		shadowOffset: {width: 5, height: 5},
		shadowRadius: 10,
		elevation: 5,
	},

	button: {
		width: '100%',
		height: '100%',
		paddingHorizontal: 20,
		paddingVertical: 5,
		justifyContent: 'center',
		alignItems: 'center',
	},

	label: {
		textAlign: 'center',
		fontSize: 16,
	},
});
