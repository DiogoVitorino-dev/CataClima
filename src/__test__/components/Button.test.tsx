import {describe,it, expect, jest, afterEach} from '@jest/globals';
import { render, fireEvent, cleanup } from '@testing-library/react-native';
import renderer from 'react-test-renderer'
import Button from '../../components/Button'

describe('Button component', () => {
  afterEach(cleanup)

  it('should renders correctly', () => {    
      const tree = renderer.create(<Button label='test' onPress={()=>{}} />).toJSON();
      expect(tree).toMatchSnapshot()    
  })
  
  it('should call a function on Pressable click', () => { 
    const fnTest = jest.fn()
    const {getByTestId} = render(<Button label='test' onPress={fnTest} />)

    fireEvent.press(getByTestId('pressableTestID'))
    expect(fnTest).toBeCalled()
  })
  
  it("shouldn't call a function if Pressable is disabled", () => { 
    const fnTest = jest.fn()
    const {getByTestId} = render(<Button label='test' onPress={fnTest} disabled />)

    fireEvent.press(getByTestId('pressableTestID'))
    expect(fnTest).not.toBeCalled()
  })
})