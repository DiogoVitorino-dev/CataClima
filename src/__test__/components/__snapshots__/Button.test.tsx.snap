// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FAFAFA",
      },
      [
        {
          "alignItems": "center",
          "borderRadius": 20,
          "elevation": 5,
          "justifyContent": "center",
          "maxHeight": 40,
          "shadowColor": "rgba(0,0,0,0.3)",
          "shadowOffset": {
            "height": 5,
            "width": 5,
          },
          "shadowRadius": 10,
        },
        undefined,
        {
          "borderColor": "rgba(0,0,0,0.3)",
          "borderWidth": 0.5,
        },
      ],
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "paddingVertical": 5,
        "width": "100%",
      }
    }
    testID="pressableTestID"
  >
    <Text
      style={
        [
          {
            "color": "#000",
          },
          [
            [
              {
                "fontSize": 16,
                "textAlign": "center",
              },
              {
                "opacity": 1,
              },
            ],
            {
              "fontFamily": "OpenSans",
            },
          ],
        ]
      }
    >
      test
    </Text>
  </View>
</View>
`;
